Program.cs

```diff
@@ -32,11 +34,19 @@ private static void Main(string[] args)
         app.MapGet("/person/GetByNameAndType", (PersonService personService, [FromQuery] string name, string emplType) 
         => Results.Ok(personService.GetPersonByNameAndPersonType(name, emplType)));
         //Products methods
+        
         app.MapGet("/product", (ProductService productService) => Results.Ok(productService.GetAll()));
         app.MapGet("/product/GetByName", (ProductService productService, [FromQuery] string name) => Results.Ok(productService.GetProductsByName(name)));
         app.MapGet("/product/GetByCatType", (ProductService productService, [FromQuery] string catType) => Results.Ok(productService.GetProductByCategoryType(catType)));
         app.MapGet("/product/GetByNameAndType", (ProductService productService, [FromQuery] string name, string catType) 
         => Results.Ok(productService.GetProductByNameAndCategoryType(name, catType)));
+        
+
+        //Sales methods
+        app.MapGet("/SalesTest",() => "Esta es una pruba");
+        app.MapGet("/sale",(SaleService salesService) => Results.Ok(salesService.GetAll()));
```

Comment:
This method presents a problem: it overflows browser memory and ends execution before completion. Implementing pagination would be beneficial. Here's an example of how to achieve pagination using C# and SQL Server in a simple manner: [Pagination in SQL Server](https://www.c-sharpcorner.com/article/pagination-in-sql-server/)

Reply:
The problem statement indeed mentions enabling users to obtain a comprehensive summary of all sales made by sales personnel. So, we were actually looking for a filtered method by person, not just reporting all the sales.